src/compile/mark.ts

```diff
@@ -69,13 +69,13 @@ export function compileMark(model: Model): any[] {
         );
 
       return [{
-        name: (name ? name + '-' : '') + mark + '-facet',
+        name: (specName ? specName + '-' : '') + mark + '-facet',
```

Comment:
Something like this? https://github.com/vega/vega-lite/blob/will/bar-label/src/compile/Model.ts#L189-L192